File: frontend/lib/models/owner.dart
class Owner {
  String? ownerId;
  String email;
  String fName;
  String lName;

  Owner({
    this.ownerId,
    required this.email,
    required this.fName,
    required this.lName,
  });

  Map<String, dynamic> toJson() {
    return {'ownerId': ownerId, 'email': email, 'fName': fName, 'lName': lName};
  }

  factory Owner.fromJson(final Map<String, dynamic> json) {
    return Owner(
        ownerId: json['ownerId'] as String?,
        email: json['email'] as String,
        fName: json['fName'] as String,
        lName: json['lName'] as String);
  }
}

File: backend_emulator/hippo-api/models/Owner.cs
using System.ComponentModel.DataAnnotations;
using Google.Cloud.Firestore;

namespace HippoApi.Models;

[FirestoreData]
public class Owner
{
    [FirestoreDocumentId] public string OwnerId { get; set; }

    [FirestoreProperty]
    [Required(ErrorMessage = "Email is required")]
    public string Email { get; set; }

    [FirestoreProperty]
    [Required(ErrorMessage = "First Name is required")]
    public string FName { get; set; }

    [FirestoreProperty]
    [Required(ErrorMessage = "Last Name is required")]
    public string LName { get; set; }
}